src/test/java/teammates/test/cases/ui/pagedata/InstructorFeedbackEditPageDataTest.java

```diff
@@ -0,0 +1,330 @@
+                                        FeedbackParticipantType.RECEIVER, 
+                                        FeedbackParticipantType.STUDENTS,
+                                        FeedbackParticipantType.OWN_TEAM_MEMBERS));
+        
+        //TODO test specific question edit form html after it is no longer html in java
+        
+        // test question add form
+        FeedbackQuestionEditForm newQuestionForm = data.getNewQnForm();
+        assertEquals(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACKS_PAGE)
+                        .withUserId(instructor.googleId)
+                        .withCourseId(fs.courseId)
+                        .withSessionName(fs.feedbackSessionName).toString(), newQuestionForm.getDoneEditingLink().toString());
+        assertFalse(newQuestionForm.getFeedbackPathSettings().isNumberOfEntitiesToGiveFeedbackToChecked());
+        assertTrue(newQuestionForm.getQuestionNumberSuffix().isEmpty());
+        
+        assertEquals(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACKS_PAGE)
+                            .withUserId(instructor.googleId)
+                            .withCourseId(fs.courseId)
+                            .withSessionName(fs.feedbackSessionName).toString(),
+                     newQuestionForm.getDoneEditingLink().toString());
+        
+        // preview form
+        FeedbackSessionPreviewForm previewForm = data.getPreviewForm();
+        assertEquals(1, previewForm.getStudentToPreviewAsOptions().size());
+        assertEquals(1, previewForm.getInstructorToPreviewAsOptions().size());
+        
+        // copy question form
+        FeedbackQuestionCopyTable copyForm = data.getCopyQnForm();
+        assertEquals(dataBundle.feedbackQuestions.size(), copyForm.getQuestionRows().size());
+        
+        
+        ______TS("empty feedback session");
+        // setup
+        data = new InstructorFeedbackEditPageData(dataBundle.accounts.get("instructor1OfCourse1"));
+        
+        fs = dataBundle.feedbackSessions.get("empty.session");
+        fs.isPublishedEmailEnabled = false;
+        fs.isClosingEmailEnabled = false;
+        
+        questions = new ArrayList<FeedbackQuestionAttributes>();
+        
+        copiableQuestions = new ArrayList<FeedbackQuestionAttributes>();
+        
+        questionHasResponses = new HashMap<String, Boolean>();
+        
+        studentList = new ArrayList<StudentAttributes>();
+        
+        instructorList = new ArrayList<InstructorAttributes>();
+        
+        instructor = getInstructorFromBundle("instructor1OfCourse1");
+        
+        data.init(fs, questions, copiableQuestions, questionHasResponses, studentList, instructorList, instructor);
+        fsForm = data.getFsForm();
+        assertEquals((new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACK_EDIT_COPY_PAGE)
+                          .withUserId(instructor.googleId)).toString(), 
+                      fsForm.getCopyToLink().toString());
```

Comment:
Several of these `get*Link()` are already returning `String`s, so calling `toString()` is pretty pointless.